docs/roadmap.md

```diff
@@ -0,0 +1,83 @@
+# Roadmap
+
+This is a list of goals we have. Many of them are quite independent.
+
+> **Note:** Most people contributing to Jujutsu do so in their spare time, which
+>  means that we cannot attach any target dates to any of the goals below.
+
+## Support for copies and renames
+
+We want to support copy tracing in a way that leaves it up to the commit backend
+to either record or detect copies. That should let us work with existing Git
+repos (Git does not record copies, it detects them on the fly) as well as with
+very large repos where detection would be too slow. See
+[design doc][copy-design-doc].
+
+## Forge integrations
+
+We would like to make it easier to work with various popular forges by providing
+something like `jj github submit`, `jj gitlab submit`, and `jj gerrit send`.
+
+## Better Rust API for UIs
+
+UIs like [gg] currently have to duplicate quite
+a bit of logic from `jj-cli`. We need to make this code not specific to the CLI
+(e.g. return status objects instead of printing messages) and move it into
+`jj-lib`.
+
+## RPC API
+
+One problem with writing tools using the Rust API is that they will only work
+with the backends they were compiled with. For example, a regular [gg] build
+will not work on Google repos because it doesn't have the backends necessary to
+load them. We want to provide an RPC API for tools that want to work with an
+unknown build of `jj` by having the tool run something like `jj api` to give it
+an address to talk to.
+
+In addition to helping with the problem of unknown backends, having an RPC API
+should make it easier for tools like VS Code that are not written in Rust. The
+RPC API will probably be at a higher abstraction level than the Rust API.
```

Comment:
I wonder if this section addresses the same problem as previous one - providing good API for third-party tools (not necessarily UI), where definition of "good" API includes working well with alternative backends. This would probably imply RPC. Not sure there is a need for separate Rust API in addition to RPC.

Reply:
FYI, I have a nice summary here https://github.com/martinvonz/jj/issues/3034#issuecomment-2030300465.

Reply:
To summarize, the RPC API will probably be at a higher abstraction level, so you won't be able to create a new command that does a totally different thing, for example. Custom `jj` binaries (like the one at Google) will also need access to the low-level parts of the API.

Reply:
Thanks, that makes sense! Do you expect other use cases where that would be needed, beyond custom `jj` binary for custom backend at Google? E.g. do you foresee alternative backends implemented outside this codebase? (Just trying to understand motivation for Rust APIs correctly.)

Not sure if that's a good direction, but potentially `jj` CLI could use RPC API instead of linking directly and Google's internal version of CLI could use additional RPC API provided by Google backend. Are there performance concerns maybe that imply the need for direct linking?

Reply:
> E.g. do you foresee alternative backends implemented outside this codebase?

I think it would make sense to add a backend for Meta's [Mononoke server[(https://github.com/facebook/sapling/tree/main/eden/mononoke) for example. Or maybe XetHub (now under Hugging Face) could implement a backend for their storage.

> Not sure if that's a good direction, but potentially `jj` CLI could use RPC API instead of linking directly and Google's internal version of CLI could use additional RPC API provided by Google backend. Are there performance concerns maybe that imply the need for direct linking?

Yes, that's a possibility.

Reply:
FYI, @ylow. I have no idea if you'd have any interest in this but I'm happy to talk about it if you want. It seems like a good match to me.